AsyncDisplayKit/ASMapNode.mm

```diff
@@ -159,6 +161,15 @@ - (void)setRegion:(MKCoordinateRegion)region
 
 - (void)takeSnapshot
 {
+  // do not trigger snapshot before layout
+  ASLayout *layout = self.calculatedLayout;
+  if (layout == nil || CGSizeEqualToSize(CGSizeZero, layout.size)) {
```

Comment:
This would mean that ASMapNode would only ever take one snapshot, since layout will never nil again after the first pass.